docs/architecture/dapr-for-net-developers/serviceinvocation.md

```diff
@@ -0,0 +1,356 @@
+            data.Items.Select(x => x.ProductId));
+
+        // Check item availability and prices; store results in basket object.
+        var basket = CreateValidatedBasket(data, catalogItems);
+
+        // Save the shopping basket.
+        await _basket.UpdateAsync(basket, authorization);
+
+        return basket;
+    }
+
+    ...
+```
+
+The `UpdateAllBasketAsync` method gets the *Authorization* header of the incoming request using a `FromHeader` attribute. The *Authorization* header contains the access token that is needed to call protected back-end services.
+
+After receiving a request to update the basket, the aggregator service calls the Catalog API to get the item details. The Basket controller uses an injected `ICatalogService` object to make that call and communicate with the Catalog API. The original implementation of the interface used gRPC to make the call. We changed the implementation to use Dapr service invocation:
+
+``` csharp
+public class CatalogService : ICatalogService
+{
+    private const string DaprAppId = "catalog-api";
+
+    private readonly DaprClient _daprClient;
+
+    public CatalogService(DaprClient daprClient)
+    {
+        _daprClient = daprClient;
+    }
+
+    public async Task<IEnumerable<CatalogItem>> GetCatalogItemsAsync(
+        IEnumerable<int> ids)
+    {
+        return await _daprClient.InvokeMethodAsync<IEnumerable<CatalogItem>>(
+            DaprAppId,
+            "api/v1/catalog/items",
+            new HTTPExtension
+            {
+                QueryString = new Dictionary<string, string>
+                {
+                    ["ids"] = string.Join(",", ids),
+                },
+                Verb = HTTPVerb.Get
+            });
+    }
+
+    ...
+}
+```
+
+Note how we use the `HTTPExtension` object to pass the query string and HTTP verb details to the Catalog API.
+
+The other call made by the aggregator service is to the Basket API. It only allows authorized requests. We pass the access token along in an *Authorization* request header to ensure the call succeeds:
+
+``` csharp
+public class BasketService : IBasketService
+{
+    private const string DaprAppId = "basket-api";
+
+    ...
+
+    public Task UpdateAsync(BasketData currentBasket, string accessToken)
+    {
+        return _daprClient.InvokeMethodAsync(
+            DaprAppId,
+            "api/v1/basket",
+            currentBasket,
+            new HTTPExtension
+            {
+                Headers = new Dictionary<string, string>
+                {
+                    [HeaderNames.Authorization] = accessToken
+                }
+            });
+    }
+
+    ...
+}
+```
+
+We don't need to explicitly specify the HTTP verb for this POST call as `POST` is the default value of `HTTPExtension.Verb`.
+
+## Summary
+
+In this chapter, we introduced the Service Invocation building block. We showed how to invoke remote methods both by making direct HTTP calls to the Dapr sidecar, and by using the Dapr .NET SDK.
```

Comment:
```suggestion
In this chapter, the service invocation building block was introduced. You saw how to invoke remote methods both by making direct HTTP calls to the Dapr sidecar, and by using the Dapr .NET SDK.
```

Reply:
Understand the need to remove "we." but wonder if sentence is now passive with "was introduced?"

Maybe, "In this chapter, you learned about about the 'blah'. You saw how to invoke 'blah'

Reply:
Rewritten to remove "we", but keeping things active